src/DynamoCore/Logging/DynamoAnalyticsClient.cs

```diff
@@ -469,7 +469,13 @@ public void Dispose()
             // If the Analytics Client was initialized, shut it down.
             // Otherwise skip this step because it would cause an exception.
             if (Service.IsInitialized)
-                Service.ShutDown();
+            {
+                // Lock shutdown sequence in case other tracking calls might be executing concurently.
+                lock(trackEventLockObj)
```

Comment:
so I might be wrong - but in all other cases this lock is held from a worker thread on the thread pool, not from main thread... could this cause a dead lock?

Reply:
From what I could tell from the [logs](https://master-5.jenkins.autodesk.com/job/Dynamo/job/DynamoSelfServe/job/pullRequestValidation/17019/testReport/junit/DynamoCoreWpfTests/CoreUserInterfaceTests/PreferenceSetting/) Looks like there is concurrent to an analytics resource (a dictionary). A dictionary is changed while it is iterated over in another thread.